services/management/grpc/actions_server.go

```diff
@@ -0,0 +1,231 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package grpc
+
+import (
+	"context"
+
+	"github.com/percona/pmm/api/managementpb"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+	"github.com/percona/pmm-managed/services/agents"
+)
+
+//nolint:unused
+type actionsServer struct {
+	r  *agents.Registry
+	db *reform.DB
+}
+
+// NewActionsServer creates Management Actions Server.
+func NewActionsServer(r *agents.Registry, db *reform.DB) managementpb.ActionsServer {
+	return &actionsServer{r, db}
+}
+
+// GetAction gets an action result.
+//nolint:lll
+func (s *actionsServer) GetAction(ctx context.Context, req *managementpb.GetActionRequest) (*managementpb.GetActionResponse, error) {
+	res, err := models.LoadActionResult(s.db.Querier, req.ActionId)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, err.Error())
+	}
+
+	return &managementpb.GetActionResponse{
+		ActionId:   res.ID,
+		PmmAgentId: res.PmmAgentID,
+		Done:       res.Done,
+		Error:      res.Error,
+		Output:     res.Output,
+	}, nil
+}
+
+// StartPTSummaryAction starts pt-summary action.
+//nolint:lll,dupl
+func (s *actionsServer) StartPTSummaryAction(ctx context.Context, req *managementpb.StartPTSummaryActionRequest) (*managementpb.StartPTSummaryActionResponse, error) {
+	a := &models.PtSummaryAction{
+		ID:                 models.GetActionUUID(),
+		NodeID:             req.NodeId,
+		PMMAgentID:         req.PmmAgentId,
+		SummarizeMounts:    true,
+		SummarizeNetwork:   true,
+		SummarizeProcesses: true,
+		Sleep:              5,
+		Help:               false,
+	}
+
+	ag, err := models.FindPMMAgentsForNode(s.db.Querier, a.NodeID)
+	if err != nil {
+		return nil, status.Errorf(codes.FailedPrecondition, err.Error())
```

Comment:
Return gRPC error returned by the helper.

Reply:
This helper and a lot of others don't return a gRPC error.

Reply:
It should (like a lot of _other_ others).

Reply:
https://github.com/percona/pmm-managed/pull/199/files#r287741912, last bullet point.